core/trino-main/src/main/java/io/trino/spiller/FileSingleStreamSpiller.java

```diff
@@ -60,8 +62,8 @@
 
     private final FileHolder targetFile;
     private final Closer closer = Closer.create();
-    private final PageSerializer serializer;
-    private final PageDeserializer deserializer;
+    private final PagesSerdeFactory serdeFactory;
```

Comment:
In commit messge you wrote:

```
Given that spilling streams are not expected to be tiny there is very
little benefit from caching serializer / deserializer.
```

If they are not tiny, then the caching is a benefit, right?
Or did you mean that a single call to `writePages` is not short?

Reply:
I mean a single call to writePages / readPages is not short, but the time spend between these two calls could in theory be long. @findepi could you please take a look and let me know if my understanding is correct?